components/server/src/auth/resource-access.ts

```diff
@@ -229,7 +233,9 @@ export namespace ScopedResourceGuard {
             case "gitpodToken":
                 return resource.subject.tokenHash;
             case "snapshot":
-                return resource.subject ? resource.subject.id : undefined;
+                return resource.subject 
+                    ? resource.subject.id 
+                    : (resource.workspaceID ? SNAPSHOT_WORKSPACE_SUBJECT_ID_PREFIX + resource.workspaceID : undefined);
```

Comment:
nit: I think it would make sense to write this out explicitly rather than as nested ternaries.
```TypeScript
if (!!resource.subject) {
    return resource.subject.id;
}
if (resource.workspaceID) {
    return SNAPSHOT_WORKSPACE_SUBJECT_ID_PREFIX + resource.workspaceID;
}
return undefined;
```